From e54839cfa32e458162bb882ad655ac82b0116d40 Mon Sep 17 00:00:00 2001
Message-Id: <e54839cfa32e458162bb882ad655ac82b0116d40.1417780878.git.chang-joon.lee@intel.com>
In-Reply-To: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
References: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
From: Hong Liu <hong.liu@intel.com>
Date: Thu, 13 Nov 2014 11:09:22 +0800
Subject: [PATCH 30/34] FOR_UPSTREAM [VPG]: drm/i915: reading correct hdmi
 audio state

Need to check correct bit for audio enabling state, otherwise we always get
following annoying message when connecting to a HDMI monitor without
audio.

<3>[   93.120951] [drm:intel_pipe_config_compare] *ERROR* mismatch in has_audio (expected 0, found 1)

Issue: GMINL-3977
Change-Id: I81bf1bc1ca97bfc1f240993046e1fa167366a8ab
Signed-off-by: Hong Liu <hong.liu@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    2 ++
 drivers/gpu/drm/i915/intel_hdmi.c    |    2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 3a71138..7b57ebf 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -11188,6 +11188,8 @@ static void intel_dump_pipe_config(struct intel_crtc *crtc,
 		      pipe_config->fdi_m_n.gmch_m, pipe_config->fdi_m_n.gmch_n,
 		      pipe_config->fdi_m_n.link_m, pipe_config->fdi_m_n.link_n,
 		      pipe_config->fdi_m_n.tu);
+	DRM_DEBUG_KMS("hdmi_sink: %i, audio: %i\n",
+		      pipe_config->has_hdmi_sink, pipe_config->has_audio);
 	DRM_DEBUG_KMS("dp: %i, gmch_m: %u, gmch_n: %u, link_m: %u, link_n: %u, tu: %u\n",
 		      pipe_config->has_dp_encoder,
 		      pipe_config->dp_m_n.gmch_m, pipe_config->dp_m_n.gmch_n,
diff --git a/drivers/gpu/drm/i915/intel_hdmi.c b/drivers/gpu/drm/i915/intel_hdmi.c
index 1d3c808..05f93f3 100644
--- a/drivers/gpu/drm/i915/intel_hdmi.c
+++ b/drivers/gpu/drm/i915/intel_hdmi.c
@@ -762,7 +762,7 @@ static void intel_hdmi_get_config(struct intel_encoder *encoder,
 	if (tmp & HDMI_MODE_SELECT_HDMI)
 		pipe_config->has_hdmi_sink = true;
 
-	if (tmp & HDMI_MODE_SELECT_HDMI)
+	if (tmp & SDVO_AUDIO_ENABLE)
 		pipe_config->has_audio = true;
 
 	pipe_config->adjusted_mode.flags |= flags;
-- 
1.7.9.5

